src/streamlink/plugins/rotana.py

```diff
@@ -5,35 +5,97 @@
 $region Saudi Arabia
 """
 
+import base64
 import logging
 import re
 
 from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
 from streamlink.stream.hls import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r'https?://(?:www\.)?rotana\.net/live'
+    r"https?://(?:www\.)?rotana\.net/"
 ))
 class Rotana(Plugin):
-    _hls_re = re.compile(r'''["'](?P<url>[^"']+\.m3u8)["']''')
+    js_url_re = re.compile(r"""['"](https://hiplayer.hibridcdn.net/l/[^'"]+)['"]""")
+    base64_data_re = re.compile(r"i\s*=\s*\[(.*)\]\.join")
 
     def _get_streams(self):
-        res = self.session.http.get(self.url)
-        m = self._hls_re.search(res.text)
-        if not m:
-            log.debug('No video URL found.')
+        js_url = self.session.http.get(
+            self.url,
+            schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'https://hiplayer.hibridcdn.net/l/')]/text()"),
+                validate.any(
+                    None,
+                    validate.all(
+                        validate.transform(self.js_url_re.search),
+                        validate.any(
+                            None,
+                            validate.all(
+                                validate.get(1),
+                                validate.url(),
+                            ),
+                        ),
+                    ),
+                ),
```

Comment:
Since there are multiple `validate.any(None, ...)` calls here for making sure that no exception is thrown, you could also flatten the validation schema and catch any `PluginError` exceptions. But ignore this for now. It's not necessary.

What we should do in the future is add an alias for this widely used pattern to the validation module.

Reply:
I've flattened L35-42 inclusive for now, as that makes sense and is easy enough.

Reply:
Can I also flatten L32-34 or might that cause linting issues in the future?

Reply:
> Can I also flatten L32-34 or might that cause linting issues in the future?

No, it's fine.